Lib/json/tool.py

```diff
@@ -15,6 +15,7 @@
 import sys
 from pathlib import Path
 
+_SENTINEL = object()
```

Comment:
This variable seems to be unnecessary.

Reply:
I used an explicit sentinel because `json.loads(line)` can return `None`, an I needed a way to distinguish the cases ("do I need to call `json.dump` on `obj` or not?")
What approach do you suggest to avoid the sentinel?

Reply:
The code `objs = (json.loads(line) if line.strip() else _SENTINEL for line in infile)` change to `objs = (json.loads(line) if line.strip() for line in infile)`,the effect is same.

Reply:
I think you mean `(json.loads(line) for line in infile if line.strip())` - as to just "ignore" empty rows in `objs`.
If I understand that well, then objs will not have any element on the newline.
Then, on #L75, on `for obj in objs: ... outfile.write('\n')` will not yield anything, not even an empty row, as output.
My desired outcome is to preserve those empty rows, so I used the `_SENTINEL`.
Or maybe are you suggesting something else?

Reply:
Oh, I didn't notice that there is a `write('\n')` in the back.

Reply:
No problem :)
Given that, I think I can mark this conversation resolved? Feel free to open it again if you have some other inputs.